hypothesis-python/src/hypothesis/control.py

```diff
@@ -125,3 +128,71 @@ def event(value):
 
     if context.data is not None:
         context.data.note_event(value)
+
+
+def target(observation, label=""):
+    # type: (float, str) -> None
+    """Calling this function with a ``float`` observation gives it a way
+    to do a directed, rather than heuristic or random, search for inputs
+    that will cause an error.  Observations must always be finite.
+
+    Hypothesis will try to drive the observed value to zero and as far from
+    zero as possible over several examples, so almost any metric will work
+    as an observation.  You can think of this as bundling togther calls to
+    imaginary ``target_zero``, ``target_positive``, and ``target_negative``
+    functions which are not available separately.
```

Comment:
I'm not a huge fan of this. I would rather have a uni-directional target function. Either always maximize, always minimize, or possibly with a `sense` argument that lets you set the direction.

I think most obvious choices of target come with a fairly natural interesting direction, and trying to optimise for two others means that we end up wasting a lot of our example budget trying to do something that doesn't matter very much.

Reply:
Just try to maximise the value for each label then? 

Makes sense, and it's easy to construct variations that do the others - I'll just write a recipe based on the current implementation!

Reply:
This conversation is marked as resolved but the docstring still shows the old behaviour?

Reply:
Just pushed the actual fix :sweat_smile: